### ISSUE TITLE
No nested schema for composed property

### ISSUE THREAD

**Issue by giann** [wol-soft/php-json-schema-model-generator#57]
**Describe the bug**
With this schema I get this error I don't understand: `No nested schema for composed property Address in file Adress.json found`

**Expected behavior**
I don't see anything wrong with this schema so I'd expect it to generate the appropriate class.

**Schema**

```json
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "oneOf": [
        { "type": "string" },
        {
            "type": "object",
            "properties": {
                "address": { "type": "string" },
                "specialDelivery": { "type": "string" },
                "zipCode": { "type": "string" },
                "city": { "type": "string" }
            },
            "additionalProperties": false
        }
    ]
}
```

No custom `GeneratorConfiguration`.

**Version:**
0.22.0

**Comment by giann** [comment]
I think I understand what's the issue. The lib doesn't seem to expect a multiple type inside a `oneOf` or `allOf` list. However I think this is a valid schema.

**Comment by wol-soft** [comment]
Hi @giann,
the lib is able to handle compositions like `oneOf` on the top level of a schema. The issue is located in the first branch of your composition which can be simplified into the following schema which makes the issue more obvious:

```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "string"
}
```

This schema can't be represented as a object. If this schema is passed to the generator it would be skipped. Due to the composition the generator starts to generate nested objects for each branch of the composition and fails on the first branch as the branch doesn't provide a nested schema which is representable as an object. I think the error message could be more meaningful.

Your schema is a valid JSON-schema but it can't be represented as a object.

**Comment by giann** [comment]
Thanks for the quick answer @wol-soft 
So i could get around this by using the oneOf in the schemas that reference this one so the oneOf is not at the top level but for, say, a property?

**Comment by wol-soft** [comment]
yes, if you wrap the composition into an object, let's say like in the following example, the generator will generate an object with the address property which is either a string or an instance of the nested object containing the properties from the second `oneOf` branch.

```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "properties": {
    "address": {
      "oneOf": [
        { "type": "string" },
        {
          "type": "object",
          "properties": {
            "address": { "type": "string" },
            "specialDelivery": { "type": "string" },
            "zipCode": { "type": "string" },
            "city": { "type": "string" }
          },
          "additionalProperties": false
        }
      ]
    }
  },
  "required": [ "address" ]
}
```

Also a reference to the schema from another schema (eg. `customer`) is possible. In this case just make sure you put the schema in let's say an `include` directory outside of the directory you have configured to generate the models from to make sure the generator doesn't try to generate a model from the address schema.

**Comment by giann** [comment]
I see thanks a lot!

**Comment by LukasGoTom** [comment]
I actually get an error as well when trying to evaluate adcp get_products json schema

original https://adcontextprotocol.org/schemas/v2/media-buy/get-products-request.json

simplified:

```
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "/schemas/2.5.0/media-buy/get-products-request.json",
  "title": "Get Products Request",
  "description": "Request parameters for discovering available advertising products",
  "type": "object",
  "properties": {
    "brand_manifest": {
      "$ref": "/schemas/2.5.0/core/brand-manifest-ref.json",
      "description": "Brand information manifest providing brand context, assets, and product catalog. Can be provided inline or as a URL reference to a hosted manifest."
    }
}
```

brand-manifest-ref.json is this:


```
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "/schemas/2.5.0/core/brand-manifest-ref.json",
  "title": "Brand Manifest Reference",
  "description": "Brand manifest provided either as an inline object or a URL string pointing to a hosted manifest",
  "oneOf": [
    {
      "$ref": "/schemas/2.5.0/core/brand-manifest.json",
      "description": "Inline brand manifest object"
    },
    {
      "type": "string",
      "format": "uri",
      "description": "URL to a hosted brand manifest JSON file. The manifest at this URL must conform to the brand-manifest.json schema."
    }
  ]
}
```

this can be simplified to

```
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$id": "/schemas/2.5.0/media-buy/get-products-request.json",
  "title": "Get Products Request",
  "description": "Request parameters for discovering available advertising products",
  "type": "object",
  "properties": {
    "brand_manifest": {
     "oneOf": [
    {
      "$ref": "/schemas/2.5.0/core/brand-manifest.json",
      "description": "Inline brand manifest object"
    },
    {
      "type": "string",
      "format": "uri",
      "description": "URL to a hosted brand manifest JSON file. The manifest at this URL must conform to the brand-manifest.json schema."
    }
  ]
    }
}
```

which actually builds just fine (praise the lord, your lib is the only one that is able to do so). I wonder though if there is a simple fix without me having to copy paste certain contents around to make it work.


so to recap: it doesnt work the way it is online with reference to a new file

it does work if i copy paste the contents of said file into the property definition.
